Packages/com.unity.inputsystem/InputSystem/Editor/Settings/InputSettingsBuildProvider.cs

```diff
@@ -18,22 +18,46 @@ public void OnPreprocessBuild(BuildReport report)
                 return;
 
             m_OriginalPreloadedAssets = PlayerSettings.GetPreloadedAssets();
+            bool wasDirty = IsPlayerSettingsDirty();
 
             if (!m_OriginalPreloadedAssets.Contains(InputSystem.settings))
             {
                 var preloadedAssets = m_OriginalPreloadedAssets.ToList();
                 preloadedAssets.Add(InputSystem.settings);
                 PlayerSettings.SetPreloadedAssets(preloadedAssets.ToArray());
             }
+
+            if (!wasDirty)
+                ClearPlayerSettingsDirtyFlag();
         }
 
         public void OnPostprocessBuild(BuildReport report)
         {
             if (InputSystem.settings == null)
                 return;
 
+            bool wasDirty = IsPlayerSettingsDirty();
+
             // Revert back to original state
             PlayerSettings.SetPreloadedAssets(m_OriginalPreloadedAssets);
+
+            if (!wasDirty)
+                ClearPlayerSettingsDirtyFlag();
+        }
+
+        static bool IsPlayerSettingsDirty()
+        {
+            var settings = Resources.FindObjectsOfTypeAll<PlayerSettings>();
```

Comment:
IIRC we have access to internals. Could probably use `PlayerSettings.GetSerializedObject` here.

There's also a `[StaticAccessor("GetPlayerSettings")]` directly on the `PlayerSettings` class. Wonder what that does...

Reply:
Turns out we don't have access after all.